core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -140,6 +142,8 @@ static Snapshot fromJson(JsonNode node) {
         }
       }
       summary = builder.build();
+      Preconditions.checkArgument(
```

Comment:
what about
```
+      operation = JsonUtil.getString(OPERATION, sNode);
+
       ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
       Iterator<String> fields = sNode.fieldNames();
       while (fields.hasNext()) {
         String field = fields.next();
-        if (field.equals(OPERATION)) {
-          operation = JsonUtil.getString(OPERATION, sNode);
-        } else {
+        if (!field.equals(OPERATION)) {
           builder.put(field, JsonUtil.getString(field, sNode));
         }
       }
```

Reply:
That works too. However, I prefer the precondition check and the error message explaining that the operation must exist.

Reply:
the issue is that the error msg should be referring to `operation` instead of `Operation`. `JsonUtil.getString(OPERATION, sNode)` will also throw a IAE and say that `operation` is missing.
Alternatively, I'd update the error msg to `Invalid summary: operation missing`

Reply:
I made the change, and the error message is now `Cannot parse missing string: operation`
Does this message give enough context? Otherwise, I can use a Precondition with a custom error message. I like `Invalid summary: operation missing`